yellow-paper/docs/contract-deployment/classes.md

```diff
@@ -16,114 +16,158 @@ Read the following discussions for additional context:
--->
+Note the lack of visibility modifiers. Internal functions are specified as a macro, and the check is handled at the application circuit level by verifying that the `context.msg_sender` equals the contract current address.
 
-### Public Function
+Also note the lack of commitment to the function compilation artifact. Even though a commitment to a function is required so that the PXE can verify the execution of correct unconstrained Brillig code embedded within private functions, this is handled entirely out of protocol. As such, PXEs are expected to verify it against the `artifact_hash` in the containing contract class.
 
-<!-- prettier-ignore -->
-| Field | Type | Description |
-|----------|----------|----------|
-| `function_selector` | `u32` | Selector of the function. Calculated as the hash of the method name and parameters. The specification of this is left to the compiler and not enforced by the protocol. |
-| `artifact_hash` | `Field` | Hash of the compiled function artifact, including all generated AVM and Brillig code. The specification of this hash is left to the compiler and not enforced by the protocol. |
-| `bytecode` | `Buffer` | An encoding of the bytecode (both avm bytecode & brillig bytecode) for the function. Must hash into the contract class's `artifact_hash`. |
+### Artifact Hash
 
-### Unconstrained Function
+Even though not enforced by the protocol, it is suggested for the `artifact_hash` to follow this general structure, in order to be compatible with the definition of the [`broadcast` function below](#broadcast).
 
-<!-- prettier-ignore -->
-| Field | Type | Description |
-|----------|----------|----------|
-| `function_selector` | `u32` | Selector of the function. Calculated as the hash of the method name and parameters. The specification of this is left to the compiler and not enforced by the protocol. |
-| `artifact_hash` | `Field` | Hash of the compiled function artifact, including all generated Brillig code. The specification of this hash is left to the compiler and not enforced by the protocol. |
-| `brillig_bytecode` | `Buffer` | An encoding of the Brillig bytecode for the function. Must hash into the contract class's `artifact_hash`. |
+```
+private_functions_artifact_leaves = artifact.private_functions.map fn => 
+  sha256(fn.selector, fn.metadata_hash, sha256(fn.bytecode))
+private_functions_artifact_tree_root = merkleize(private_functions_artifact_leaves)
 
-<!-- TODO(palla): Expand on the bytecode commitment scheme and bytecode_hash, both here and for private fns. -->
+unconstrained_functions_artifact_leaves = artifact.unconstrained_functions.map fn => 
+  sha256(fn.selector, fn.metadata_hash, sha256(fn.bytecode))
+unconstrained_functions_artifact_tree_root = merkleize(unconstrained_functions_artifact_leaves)
 
-### Class Identifier
+version = 1
+artifact_hash = sha256(
+  version, 
```

Comment:
Just removed that version field, since it was unnecessary. The artifact currently includes name and version of compiler (`"noir_version": "0.23.0+e470b94de7f3a0db4999a4b57d6e97298505415c"`). As for vks, they are emitted one for each function.

I don't know about the proving system info though, I'll check!